src/odin/maneuversbuilder.cc

```diff
@@ -2354,15 +2424,13 @@ bool ManeuversBuilder::IsFork(int node_index,
       return false;
     };
 
-    // The curr edge and intersecting edge must be highway or ramp but not both
-    // validate lane bifurcation
     if (prev_edge->IsHighway() &&
         ((curr_edge->IsHighway() && (xedge->use() == TripLeg_Use_kRampUse)) ||
          (xedge->IsHighway() && curr_edge->IsRampUse())) &&
+        has_lane_bifurcation(trip_path_, node_index, prev_edge, curr_edge, xedge) &&
```

Comment:
Checking for the lane-bifurcation now has precedence over the fork-forward angle checks.